### PR TITLE
feat: add send email verification button

### PR DESCRIPTION
**Pull request by Chisomchima**
Implements [DHIS2-18374](https://dhis2.atlassian.net/browse/DHIS2-18374)
---

### Description
This PR introduces functionality to allow users to initialize the email verification process from their profile page. It adheres to the outlined acceptance criteria and integrates backend communication to handle email verification requests.

---

### Checklist

-   [ ] Added a "Verify Email" button to the user profile page, which is displayed only under specific conditions.
-   [ ] Integrated API call to the backend verify email endpoint (POST /api/account/sendEmailVerification).
-   [ ] Implemented UI feedback (spinner and disabled state) during request progress.
-   [ ] Handled request success and failure states, displaying appropriate alert messages.

[DHIS2-18374]: https://dhis2.atlassian.net/browse/DHIS2-18374?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/layout/FormFields.component.js [hunk 1]
```diff
@@ -345,15 +351,33 @@ class FormFields extends Component {
...
-
+        // Create the regular fields
         const fields = fieldNames
             .map((fieldName) => createField(fieldName, valueStore, d2))
             .filter((field) => !!field.name)
             .map((field) => wrapFieldWithLabel(field))
 
+        // Append the Verify Email button after the 'email' field
+        const emailFieldIndex = fieldNames.indexOf('email')
+        if (emailFieldIndex !== -1) {
```
Comment (tomzemp) [inline src/layout/FormFields.component.js:369]: Is there a particular reason you're inserting the field with this splice operation?

I would think you should be able to add `emailVerification` after `email` in the fieldKeys in EditProfile, and then you could define a createEmailVerificationField() function that returns the component you wrote 🤔 (e.g see the definition of `createAvatarEditor` for a similar approach that also loads a functional component and passes some values from valueStore as props)

**Review by tomzemp — CHANGES_REQUESTED** [review]
Looks good 😄! 

I left a couple of suggestions / questions. In terms of functionality, I think just fixing the emailConfigured and the conditional display of the "Verify email" button (i.e. hide the button when email is not configured) are necessary to satisfy the ticket requirements.

File: src/layout/FormFields.component.js [hunk 2]
```diff
@@ -345,15 +351,33 @@ class FormFields extends Component {
         this.disposable.unsubscribe()
     }
 
+    handleEmailVerification = async () => {
```
[the remark this answers is not in the document]
Comment (Chisomchima) [reply src/layout/FormFields.component.js:354]: will be removed once using useDataMutation

File: src/layout/FormFields.component.js (hunk 1 above)
Comment (Chisomchima) [reply src/layout/FormFields.component.js]: great catch... will be considered in the next commit.

**Review by tomzemp — APPROVED** [review]
Looks good! 🚀

Do you want to remove the `handleEmailVerification` from src/layout/FormFields.component.js as it is not used?

**Pull request merged**